### DISCUSSION TITLE
WPF render problems on i5-11600 (Intel UHD Graphics 750)

### DISCUSSION THREAD

**Discussion by che6urek** [cefsharp/CefSharp#3806] [General]
- **What version of the product are you using?**
    93.1.111, 92.0.260 (**67.0.0 works fine**)

- **What architecture x86 or x64?**
    x86, x64, AnuCPU
	
- **What version of .Net?**
    .Net 4.5.2, Net 4.6.2
    
- **On what operating system?**
    Win10

- **Are you using `WinForms`, `WPF` or `OffScreen`?**
    WPF (WinForms, OffScreen work fine)
    
- **What steps will reproduce the problem?**
  - Run MinimalExample WPF.
  - Type into google search bar.
  - Nothing happen.

    Latest versions of CefSharp WPF work very poorly on the i5-11600 (driver with [WPF fix](https://www.intel.com/content/www/us/en/support/articles/000058136/graphics/graphics-for-11th-generation-intel-processors.html) is installed, any other WPF works perfectly fine). I've tested it on different PCs, so it's most likely not an environment issue. It works fine with WinForms and OffScreen as well as WPF version 67.0.0. It works fine with the latest WPF version on all other CPUs I've tried (older ones), and on i5-11600 via RPD with another CPU. This is a rendering issue, when I type something in the google search bar, the input is not shown. But after resizing the window, the input appears as well as the dropdown. There is only one error message in the output on the start `[0913/174644.236:ERROR:crashpad_client_win.cc(520)] CreateProcess: The system cannot find the file specified. (0x2)`. But there is the same message for WinForms and everething works fine.

    For my application, I am using our internal web controls and they are not showing at all. I tried one weird fix: set a timer to resize the window by 1 pixel every 100ms in order to call the render operation. Of course I can't use this fix, but in this case it renders everything fine.

   Yep, I understand that this is a hardware problem, but version 67.0.0 is working fine, so I hope there is something that can be fixed.

- **Does this problem also occur in the `CEF` Sample Application**
   No

**Comment by amaitland** [comment]
> . It works fine with the latest WPF version on all other CPUs I've tried (older ones), and on i5-11600 via RPD with another CPU.

Converting to discussion as this is unlikely a bug in CefSharp as it works correctly on every other platform tested. Sounds like a problem with the Intel driver. Did you try the disabling WPF hardware acceleration? 

https://github.com/cefsharp/CefSharp/issues/3280#issue-742243962

> [0913/174644.236:ERROR:crashpad_client_win.cc(520)] CreateProcess: The system cannot find the file specified. (0x2). But there is the same message for WinForms and everething works fine.

Likely safe to ignore as crash reporting is unrelated to rendering.

**Reply by che6urek** [reply]
Disabling WPF hardware acceleration almost fixed the problem. Thank you. But it still works a little bit slower than the 67.0.0. Animations are more laggy. Why is it working on an old version without additional fixes? There isn't hardware acceleration?

**Reply by amaitland** [reply]
> Disabling WPF hardware acceleration almost fixed the problem.

Triple check you have the correct driver. If you are confident you have the correct driver then you should report the problem to `Intel` as there is likely a problem they need to resolve.

https://community.intel.com/t5/Graphics/bd-p/graphics

> Thank you. But it still works a little bit slower than the 67.0.0. Animations are more laggy. Why is it working on an old version without additional fixes? There isn't hardware acceleration?

Best guess would be that the older [InteropBitmap](https://docs.microsoft.com/en-us/dotnet/api/system.windows.interop.interopbitmap?view=net-5.0) implementation that was used by default previously isn't affected by the bug, the newer [WritableBitmap](https://docs.microsoft.com/en-us/dotnet/api/system.windows.media.imaging.writeablebitmap?view=net-5.0) implementation has the problem as detailed in https://github.com/dotnet/wpf/issues/3817

- https://github.com/cefsharp/CefSharp/blob/cefsharp/93/CefSharp.Wpf/ChromiumWebBrowser.cs#L2651
- https://github.com/cefsharp/CefSharp/blob/cefsharp/67/CefSharp.Wpf/ChromiumWebBrowser.cs#L1601


You should still be able to use the older implementation with something like the following:
```c#
browser.RenderHandler = new CefSharp.Wpf.Rendering.InteropBitmapRenderHandler();
```

The newest hardware I personally own is 8+ years old, so this is not something I can test myself.
